.github/workflows/mypy.yml

```diff
@@ -0,0 +1,26 @@
+name: Run mypy
```

Comment:
I think we have a pre-commit bot that already runs on PRs, so in theory should be running the added mypy to the `.pre-commit-config.yaml`

Reply:
You are right, Now I can see it is triggered in `.github/workflows/check-pr.yml`.
What is unclear to me why is it failing on the pre-commit.ci server. It seems that it does not take in account the `.mypy.ini` that excludes the duplicate files.

Reply:
I've commented out mypy in the pre-commit hook to make ti pass on the pre-commit.ci server.
We still have it on GitHub Actions.
Either someone will figure out what's the problem on the pre-commit.ci server or this can be revisited later.